tests/Rust/tests/src/AsyncTests.fs

```diff
@@ -93,30 +93,53 @@ let shouldExecuteMutationOnTask () =
     do comp.Result
     x |> equal 1
 
+// open System.Threading
 // [<Fact>]
 // let ``should execute mutation on thread unsafe`` () =
 //     let mutable x = 1
-//     let t = new System.Threading.Thread(fun () -> x <- x + 1)
+//     let t = Thread(fun () -> x <- x + 1)
 //     t.Start()
 //     t.Join()
 //     x |> equal 2
 
-// module Monitor =
-//     [<Fact>]
-//     let monitorShouldWorkWithSystemObj () =
-//         let o = new System.Object()
-//         System.Threading.Monitor.Enter(o)
-//         System.Threading.Monitor.Exit(o)
+module Monitor =
+    // [<Fact>]
+    // let monitorShouldWorkWithSystemObj () =
+    //     let o = new System.Object() // todo - this doesn't work, and outputs unit
```

Comment:
@alexswan10k Yeah, looks like in `Rust/Replacements` (and elsewhere) this gets converted to an object expression, but it doesn't have to be like that, it can be whatever we want it to be. Please also keep in mind that currently the `obj` type (i.e. `System.Object`) is erased to `Fable.Any`, but that may not be ideal either, see discussion in #3027.